### DISCUSSION TITLE
How to query subfolder of content

### DISCUSSION THREAD

**Discussion by BD103** [nuxt/content#2127] [Q&A]
Hello,

I would like all of my content to be prefixed with the `/blog` route. My current solution is to store all posts in a `content/blog` folder.

The problem that I'm experiencing is that when I use `queryContent()`, it treats the folder as a post.

```ts
const posts = await fetchContentNavigation(queryContent("blog"));
```

```json
[
  {
    "title": "Blog",
    "_path": "/blog",
    "children": [
      {
        "title": "Foo",
        "_path": "/blog/foo"
      },
      {
        "title": "Bar",
        "_path": "/blog/bar"
      }
    ]
  }
]
```

Is there a way that I could just query the `children` object while still using `queryContent()`'s functions like `sort()`, `limit()`, and `skip()`?

Thank you :)

**Edit:**

Here's a reproduction of my issue:

https://stackblitz.com/edit/github-zvmcby-u8utm2?file=app.vue

What I don't want:

```json
[
  {
    "title": "Blog",
    "_path": "/blog",
    "children": [
      {
        "title": "Bar",
        "_path": "/blog/bar"
      }
    ]
  }
]
```

What I want:

```json
[
  {
    "title": "Bar",
    "_path": "/blog/bar"
  }
]
```

**Comment by nobkd** [comment]
You can query by regex.
```ts
queryContent().where({ _path: /^\/blog\// })
```
This gets all posts that have a path that starts with `/blog/`.

Similar reference: https://github.com/nuxt/content/discussions/1777#discussioncomment-4502897

Only thing is, that I don't know if the root path can be `/blog/`.
If that is the case, you can try something like this regex instead: `/^\/blog\/.+/`.
I haven't tested it, but that should probably work. You can just edit the regex to fit your needs.

**Reply by BD103** [reply]
I tried your solution, but I can't get regex to work. I tried `queryContent().where({ _path: /a+/ })` to deliberately not match `/blog`, but it still returns the same JSON as above. Any other solutions?

**Reply by nobkd** [reply]
Hm. It seems like `fetchContentNavigation()` can't take just any `queryContent()` input?

Do you know something about that, @farnabaz ?

The docs state, you could put [`Any query built via queryContent()`](https://content.nuxtjs.org/api/composables/fetch-content-navigation#arguments) in there, but that seems to be not correct?

An example: https://stackblitz.com/edit/github-zvmcby-yt8pwh?file=app.vue

**Reply by BD103** [reply]
I found [this section](https://content.nuxtjs.org/guide/displaying/navigation#nested-navigation) in the docs, but it appears to have an outdated example.

```ts
const query = queryContent({
  where: {
    _path: { $contains: '/your/navigation/starting/point' }
  }
})
```

I tried both it and chaining `queryContent().where(...)`, and neither worked. (The latter gave the nested JSON, while the first didn't run.)

**Comment by BD103** [comment]
I [crossposted this on Discord](https://discord.com/channels/473401852243869706/713346822143475752/1123949483990982667) to get a few more eyes on the situation.
